File: backend/src/main/java/org/example/backend/domain/tradechat/entity/TradeChatMessage.java
package org.example.backend.domain.tradechat.entity;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.example.backend.domain.member.entity.Member;

import java.time.LocalDateTime;

@Entity
@Table(name = "trade_chat_message")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class TradeChatMessage {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column (name = "message_id", nullable = false)
    private Long Id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn (name = "room_id", nullable = false)
    private TradeChatRoom chatRoom;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn (name = "sender_id", nullable = false)
    private Member sender;

    @Column(nullable = false)
    private String content;

    @Column(nullable = false)
    private LocalDateTime sendDate;

    // 메세지 채팅 생성
    public static TradeChatMessage create(TradeChatRoom chatRoom, Member sender, String content) {
        TradeChatMessage message = new TradeChatMessage();
        message.chatRoom = chatRoom;
        message.sender = sender;
        message.content = content;
        message.sendDate = LocalDateTime.now();
        return message;
    }
}

File: backend/src/main/java/org/example/backend/domain/tradechat/entity/TradeChatMessage.kt
package org.example.backend.domain.tradechat.entity

import jakarta.persistence.*
import org.example.backend.domain.member.entity.Member
import java.time.LocalDateTime

@Entity
@Table(name = "trade_chat_message")
class TradeChatMessage private constructor(
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "room_id", nullable = false)
    val chatRoom: TradeChatRoom,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "sender_id", nullable = false)
    val sender: Member,

    @Column(nullable = false)
    val content: String,

    @Column(nullable = false)
    val sendDate: LocalDateTime = LocalDateTime.now()
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0L
        private set

    companion object {
        // 메세지 채팅 생성
        fun create(chatRoom: TradeChatRoom, sender: Member, content: String): TradeChatMessage =
            TradeChatMessage(
                chatRoom = chatRoom,
                sender = sender,
                content = content
            )
    }
}
